zio-http/src/test/scala/zhttp/middleware/MiddlewareSpec.scala

```diff
@@ -166,6 +166,52 @@ object MiddlewareSpec extends DefaultRunnableSpec with HttpAppTestExtensions {
               res <- app(request)
             } yield assert(res.getHeadersAsList)(hasSubset(expected))
           }
+      } +
+      suite("addCookie middleware") {
+        testM("should add set-cookie header") {
+          val app = (Http.ok @@ addCookie(Cookie("test", "testValue"))).getHeader("set-cookie")
+          assertM(app(Request()))(
+            equalTo(Some(Cookie("test", "testValue").encode)),
+          )
+        } +
+          testM("should add set cookie header with value produced by effect") {
+            val app =
+              (Http.ok @@ addCookieM(UIO(Cookie("test", "testValue")))).getHeader("set-cookie")
+            assertM(app(Request()))(
+              equalTo(Some(Cookie("test", "testValue").encode)),
+            )
+          }
+      } +
+      suite("CSRF middleware") {
+        val app: Http[Any, Nothing, Request, Status] = (Http.ok @@ csrf("x-token")).getStatus
+        val cookieHeader                             = Headers.cookie(Cookie("x-token", "secret"))
+        testM("should give forbidden if token is not present in header") {
+          assertM(app(Request(headers = cookieHeader)))(equalTo(Status.FORBIDDEN))
+        } +
+          testM("should give forbidden if token is present in header but doesn't match with token cookie") {
+            assertM(app(Request(headers = cookieHeader ++ Headers("x-token", "secret1"))))(
+              equalTo(Status.FORBIDDEN),
+            )
+          } +
+          testM("should give OK if token present in header matches token present in cookie") {
+            assertM(app(Request(headers = cookieHeader ++ Headers("x-token", "secret"))))(
+              equalTo(Status.OK),
+            )
+          } +
+          testM("should give empty body if token is present in header but doesn't match with token cookie") {
```

Comment:
```suggestion
          testM("empty body") {
```

Can even drop this test